README.md

```diff
@@ -0,0 +1,116 @@
+# **ALTERNATE BIT PROTOCOL USING CADMIUM**
+---
+### ORGANINIZATION
+---
+#### Carleton University
+### AUTHORS
+---
+**Originally Developed by**: Cristina Ruiz Martin<br/>
+**Modified by**: Peter Bliznyuk-Kvitko and Shubham Agrawal
+### INTRODUCTION
+---
+ABP (Alternating Bit Protocol) is a communication protocol to ensure reliable transmission through unreliable network. In ABP protocol sender sends a packet and waits for an acknowledgement. If the acknowledgement doesn't arrive within a predefined time, the sender re-sends this packet until it receives an expected acknowledgement and then sends the next packet. In order to distinguish two consecutive packets, the sender adds an additional bit on each packet (called alternating bit because the sender uses 0 and 1 alternatively). 
+
+This software is a simulator to replicate the behavior of the Alternating Bit Protocol. The ABP simulator consists of 3 components.
+
+- **Receiver** - It receives the data and sends back acknowledgement after certain perios of time.
+- **Subnets** - It just transmits the packets to next node after a time delay. 
+- **Sender** - It sends the packet and then waits for acknowledgement from the receiver but if it does not receive the acknowledgement within time window it will resend the packet with alternating bit attached to it so that it can differentiate between the two packets. Once it receives the acknowledgement it will swnd the next packet until it has no more packets to send.
+### FILES ORGANIZATION
+---
+##### data [This folder contains the data files for the simulator]
+1. input
+    -   input_abp_0.txt
+    -    input_abp_1.txt
+2. output
+	-	-abp_output_0.txt
```

Comment:
The hypen ( - ) before the file names can be removed as it not followed in other sections of the file

Reply:
solved